src/DataAccess/ApiBasedPageRetriever.php

```
@@ -28,14 +28,24 @@ public function __construct( MediawikiApi $api, ApiUser $apiUser, LoggerInterfac
 		$this->logger = $logger;
 	}
 
-	public function fetchPage( string $pageTitle ): string {
+	public function fetchPage( string $pageTitle, string $action = 'render' ): string {
 		$this->logger->debug( __METHOD__ . ': pageTitle', [ $pageTitle ] );
 
 		if ( !$this->api->isLoggedin() ) {
 			$this->doLogin();
 		}
 
-		$content = $this->retrieveRenderedPage( $pageTitle );
+		$content = false;
+		switch ( $action ) {
+			case 'raw':
+				$content = $this->retrieveWikiText( $pageTitle );
+				break;
+			case 'render':
+				$content = $this->retrieveRenderedPage( $pageTitle );
+				break;
+			default:
+				break;
```

Should't this fail loudly with a RuntimeException?